### PR TITLE
CRM-17789 Fix mysqli::free and update DB_DataObject free to what is in their rep…

### PR DESCRIPTION
**Pull request by seamuslee001**
…o and port mollux change to DB.php for free

### CODE REVIEW

**Comment by seamuslee001** [comment]
@totten @mollux this is a resubmit of https://github.com/civicrm/civicrm-packages/pull/149 Note i have made slightly different changes to the free() in DB_DataObject. They match up to the source here http://svn.php.net/viewvc/pear/packages/DB_DataObject/trunk/DB/DataObject.php?revision=339312&view=markup

**Comment by totten** [comment]
Did a test run on `test-ubu1204-5` with `pkgs-169-1` (`CRM`, `api_v3`, `Civi`, and `civicrm-upgrade-test` w/4.2.0 or 4.4.0). All those tests passed, except `CRM_Contribute_Form_ContributionTest::testSubmitCreditCardPayPal` which has been ornery lately.

**Comment by seamuslee001** [comment]
@totten so we can merge?

File: DB/mysqli.php [hunk 1]
```diff
@@ -498,7 +498,11 @@ function fetchInto($result, &$arr, $fetchmode, $rownum = null)
      */
     function freeResult($result)
     {
-        return is_resource($result) ? mysqli_free_result($result) : false;
```
Comment (eileenmcnaughton) [inline DB/mysqli.php]: I have had to revert this for wmf to get it to pass their tests . I'm not sure if the problem is on the servers - or only the test env (running hhvm) - https://integration.wikimedia.org/ci/job/wikimedia-fundraising-civicrm/3756/console

Comment (seamuslee001) [inline DB/mysqli.php]: Feels strange this changes matches the core project tho

Comment (eileenmcnaughton) [inline DB/mysqli.php]: I didn't actually find the core file - the link above doesn't have that line in it

Comment (seamuslee001) [inline DB/mysqli.php]: see https://github.com/pear/DB/blob/trunk/DB/mysqli.php#L498

**Pull request merged**